### DISCUSSION TITLE
Can not compile on Arduino Micro

### DISCUSSION THREAD

**Discussion by frankmartindev** [NicoHood/HID#389] [Q&A]
Hi,
I am trying to "build" remote controler for my pc. So far I defined couple keys, but I need some multimedia keys. 
When I include a necessary libs and trying to compile I got an error. Could you explain that?

**Comment by rokzin84** [comment]
What is the error?

**Reply by frankmartindev** [reply]
"exit status 1
Błąd kompilacji dla płytki Arduino Micro."

**Reply by rokzin84** [reply]
Are the libs compatible with the Arduino Micro?
Can you copy and paste the entire error message here?

**Reply by frankmartindev** [reply]
Sure.

Arduino: 1.8.19 (Windows 10), Board: "Arduino Leonardo"

In file included from c:\users\marci\onedrive\documents\arduino\libraries\hid-project\src\hid-apis\KeyboardAPI.h:29:0,

                 from c:\users\marci\onedrive\documents\arduino\libraries\hid-project\src\hid-apis\defaultkeyboardapi.h:27,

                 from C:\Users\marci\OneDrive\Documents\Arduino\libraries\HID-Project\src/SingleReport/BootKeyboard.h:30,

                 from C:\Users\marci\OneDrive\Documents\Arduino\libraries\HID-Project\src/HID-Project.h:50,

                 from C:\Users\marci\OneDrive\Documents\Arduino\ir_remote_keyboard\ir_remote_keyboard.ino:1:

c:\users\marci\onedrive\documents\arduino\libraries\hid-project\src\keyboardlayouts\improvedkeylayouts.h:54:21: note: #pragma message: Using default ASCII layout for keyboard modules

     #pragma message "Using default ASCII layout for keyboard modules"

                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In file included from C:\Users\marci\OneDrive\Documents\Arduino\ir_remote_keyboard\ir_remote_keyboard.ino:4:0:

C:\Users\marci\OneDrive\Documents\Arduino\libraries\Keyboard\src/Keyboard.h:130:7: error: redefinition of 'class Keyboard_'

 class Keyboard_ : public Print

       ^~~~~~~~~

In file included from C:\Users\marci\OneDrive\Documents\Arduino\libraries\HID-Project\src/HID-Project.h:51:0,

                 from C:\Users\marci\OneDrive\Documents\Arduino\ir_remote_keyboard\ir_remote_keyboard.ino:1:

C:\Users\marci\OneDrive\Documents\Arduino\libraries\HID-Project\src/MultiReport/ImprovedKeyboard.h:33:7: note: previous definition of 'class Keyboard_'

 class Keyboard_ : public DefaultKeyboardAPI

       ^~~~~~~~~

Multiple libraries were found for "Keyboard.h"

 Used: C:\Users\marci\OneDrive\Documents\Arduino\libraries\Keyboard

 Not used: C:\Program Files (x86)\Arduino\libraries\Keyboard

exit status 1

Error compiling for board Arduino Leonardo.



This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

**Reply by rokzin84** [reply]
I think the problem is defining class Keyboard_ in the different files.
I am not intimate with the different keyboard files, but you could try to remove the keyboard definitions from the files so that you are only left with 1 definition.
It seems the error is because class Keyboard_ has been defined several places, they are trying to fight each other, but the compiler sees it and generates the error.

The line numbers are next to the files, check them out and try commenting out some class Keyboard_

**Reply by frankmartindev** [reply]
Holy bananas! I was blind. I will try when I come back from work. Thanks!

**Reply by frankmartindev** [reply]
@rokzin84 could you help me with `#define MEDIA_VOLUME_MUTE	0xE2` How to use it? I have old remote control from HP laptop and now I want to use it with current laptop.  For example one of button on remote is defined like this:
`      case runExplorer:
        Keyboard.press(KEY_LEFT_GUI);
        Keyboard.press('e');
      break;`

**Reply by frankmartindev** [reply]
Nevermind, I found the solution.

**Reply by rokzin84** [reply]
Awesome.
Did your project work after removing the #defines?
I am new to programming myself, but eager to learn.
Trying to use this project to make myself a gamepad working with fortnite on an analogue stick.

**Reply by frankmartindev** [reply]
Hi, here is a link to my project. https://github.com/mapastudiopl/arduino/blob/main/remote_pc.ino
